packages/@aws-cdk/aws-ecs-patterns/lib/base/queue-processing-service-base.ts

```diff
@@ -136,6 +160,11 @@ export abstract class QueueProcessingServiceBase extends Construct {
   constructor(scope: Construct, id: string, props: QueueProcessingServiceBaseProps) {
     super(scope, id);
 
```

Comment:
Feels like we can probably reuse this logic across patterns (via common utility function)

Reply:
Potentially, I'll leave this PR as is, and create an issue to discuss this, there's a lot of common functionality that's separated today.